### ISSUE TITLE
Rule proposal: `consistent-conditional-object-spread`

### ISSUE THREAD

**Issue by zanminkian** [sindresorhus/eslint-plugin-unicorn#2363]
### Description

When spreading an object conditionally, we have two style: `? :` and `&&`. I prefer user consistently to use `&&`.

### Fail

```ts
const name = Math.random() > 0.5 'foo' : undefined;
const student = {age: 12, ...(name&&{name})};
```

### Pass

```ts
const name = Math.random() > 0.5 'foo' : undefined;
const student = {age: 12, ...(name ? {name} : {})};
```

### Proposed rule name

consistent-conditional-object-spread

**Comment by sindresorhus** [comment]
I prefer `&&` too in this case. There is really no benefit of the verbose ternary version. However, it should include an option to choose the style.

**Comment by fisker** [comment]
ONE benefit in werid JavaScript world.

```js
({...(document.all && {foo: 1})})

// VS

({...(document.all ? {foo: 1} : undefined)})
```

Just kinding.

**Comment by zanminkian** [comment]
From my point of view, this rule only work when the variable name is the same as the property name. For example:

Work:
```js
const student = {...(name?{name}:{})}
// or
const student = {...(name&&{name})}
```

Not work:
```js
const student = {...(name?{age}:{})}
// or
const student = {...(name&&{age})}
```

**Comment by fisker** [comment]
> this rule only work when the variable name is the same as the property name. 

I don't think so, the name doesn't matter. Spread any falsely value in object literal does not add any property (of cause except HTMLDDA).

**Comment by sindresorhus** [comment]
Accepted

**Comment by sindresorhus** [comment]
Fixed by https://github.com/sindresorhus/eslint-plugin-unicorn/commit/8aa1e0de1c36a6e0df6fc1181b3b674126d11dc7
